### ISSUE TITLE
Failed to Compile on Void musl

### ISSUE THREAD

**Issue by g-h-97** [yshui/picom#872]
### Platform
Voidlinux musl

### Environment
dwm

### picom version
Latest git version

### Configuration:
None

### Steps of reproduction
Compile picom under Void musl

### Expected behavior
`picom` binary under src/

### Current Behavior
Linker failure

### Other details
Error text file, [error.txt](https://github.com/yshui/picom/files/9353057/error.txt)

**Comment by yshui** [comment]
Can you report this to void upstream? Likely their packaging of libconfig is broken

**Comment by g-h-97** [comment]
Here is the response,

> xbps-src picom builds. Maybe their broke their build script since release, or maybe you haven't install everything required (pkg-config?), hard to say without build log.

Dev, could you please assist on collecting the build log ?

**Comment by tryone144** [comment]
The GCC command line does not include any flags to actually link to `libconfig`.
If you have followed [the readme](https://github.com/yshui/picom#to-build), check that `ninja -C build reconfigure` actually finds the libconfig dependency.

**Comment by yshui** [comment]
meson would have failed if it can't find libconfig.

@g-h-97 right, try `reconfigure` is a good idea. also, what's the output of `pkg-config --libs libconfig`?

**Comment by g-h-97** [comment]
@tryone144, I did everything on the readme as far as I can tell. Find bellow the output of `ninja -C build reconfigure`:
[ninja_recon.txt](https://github.com/yshui/picom/files/9357199/ninja_recon.txt)

@yshui, please find the output of the `reconfigure` command above, also here is the output of `pkg-config --libs libconfig`:
`-lconfig`

**Comment by 0x5c** [comment]
The template builds without issues for the latest commit (ec9243cf3c824ce6f910ef5800c4af185f12721b).

Here's the changes, compared to the [current version of the template](https://github.com/void-linux/void-packages/blob/137ff8ef506c64eb269e3bb7c8a4f589a75f4017/srcpkgs/picom/template). All of those changes are only related to getting the specific commit and ensuring that xbps doesn't think this package is plain picom 9.1. 
```diff
diff --git i/srcpkgs/picom/template w/srcpkgs/picom/template
index 8c35218384..4c7ae88ec1 100644
--- i/srcpkgs/picom/template
+++ w/srcpkgs/picom/template
@@ -1,8 +1,10 @@
 # Template file for 'picom'
 pkgname=picom
-version=9.1
+_commit="ec9243cf3c824ce6f910ef5800c4af185f12721b"
+version=9.1+1
 revision=1
 build_style=meson
+wrksrc="picom-$_commit"
 configure_args="-Dwith_docs=true"
 hostmakedepends="pkg-config asciidoc"
 makedepends="MesaLib-devel dbus-devel libconfig-devel libev-devel pcre-devel
@@ -13,8 +15,8 @@ short_desc="Lightweight compositor for X11 (previously a compton fork)"
 maintainer="Orphaned <orphan@voidlinux.org>"
 license="MIT, MPL-2.0"
 homepage="https://github.com/yshui/picom/"
-distfiles="https://github.com/yshui/picom/archive/v${version}.tar.gz"
-checksum=8700ac71bd496c91262c8576e29cb3aecf2b4ef48c04394a929509d3cb37b87d
+distfiles="https://github.com/yshui/picom/archive/${_commit}.tar.gz"
+checksum=6a46d02521bdb86c02f2ed66a0fe4c33c0cdd6fc3de892fc911ffc621217c908
```

Since templates are run in an environment controlled by xbps-src, being able to build inside but not outside of it indicates something is not done right when attempting to build manually outside of xbps-src.
Since the template already fine and xbps-src guarantees a proper build, I recommend you try it

**Comment by yshui** [comment]
@g-h-97 So looks like the latest git commit is able to be built by xbps-src, this is neither picom or void's problem.

It's difficult to debug problems like this over the Internet, basically you need to figure out why meson didn't add `-lconfig` to your link flags.

**Comment by g-h-97** [comment]
@yshui I'm trying to compile the `pijulius` for of picom https://github.com/pijulius/picom, I noticed that it points to this repo for opening issues. I'm certain that it has a relatively different compile procedure, as such I made a `xbps-src` template to compile  `vNext` branch which failed to compile as well.

This makes me confident that the issue here is not related to `libconf` per se, rather it has to do with the void system being based on `musl` libc.

I ended up switching to Arch and installing `picom-pijulius-git` AUR package instead which is working wonders for now.The main thing I want to point out is, why the pijulius fork is pointing `issues` section to this repo ?

Finally, I truly appreciate you help guys, and I'm truly sorry for any misunderstanding, Closing this case for now but I would appreciate answering my above question,

Thank you,
